src/book.cpp

```diff
@@ -72,9 +78,10 @@ bool Book::update(const kiwix::Book& other)
   m_articleCount = other.m_articleCount;
   m_mediaCount = other.m_mediaCount;
   m_size = other.m_size;
-  m_favicon = other.m_favicon;
-  m_faviconMimeType = other.m_faviconMimeType;
-  m_faviconUrl = other.m_faviconUrl;
+  m_illustrations.clear();
+  for ( const auto& ill : other.m_illustrations ) {
+    m_illustrations.push_back(std::make_shared<Illustration>(*ill));
+  }
```

Comment:
I wonder if we still need this.
As we are storing `std::shared<const Illustration>` it should be ok to share the illustration between several books.
And so we can use the copy constructor.

Reply:
We don't store `std::shared_ptr<const Illustration>` in `m_illustrations` (we use `std::shared_ptr<Illustration>` instead). `Book::getMutableDefaultIllustration()` may return a reference to an existing entry in `m_illustrations`. The `const_cast` therein is justified by the fact that the const `getDefaultIllustration()` member function is called from a non-const member function.

Sharing the default illustration between books will misbehave when a book is copied from another via `Book::update()`, and then one of them is updated by `Book::updateFromXml()` or `Book::updateFromOpds()` - the default illustration will change in both books. Though this scenario doesn't seem likely, we better not open up possibilities for such bugs.

Reply:
> Sharing the default illustration between books will misbehave when a book is copied from another via Book::update(), and then one of them is updated by Book::updateFromXml() or Book::updateFromOpds() - the default illustration will change in both books. Though this scenario doesn't seem likely, we better not open up possibilities for such bugs.

In my idea, all illustrations are const. So a `Book::updateFrom*` always clear the `m_illustrations` vector and recreate the illustration(s) from what is in the xml/opds/archive. We don' share the default illustration because there would be no default illustration.

Reply:
Deal